src/Passes/Source/Validator/Validator.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "AllocationManager/AllocationManager.hpp"
+#include "AllocationManager/IAllocationManager.hpp"
+#include "ValidationPass/ValidationConfiguration.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+#include <memory>
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    /// Validator class that defines a set of rules which constitutes the profile definition. Each of
+    /// the rules can be used to transform a generic QIR and/or validate that the QIR is compliant with
+    /// said rule.
+    class Validator
+    {
+      public:
+        using ValidatorPtr = std::unique_ptr<Validator>;
+
+        // Constructors
+        //
+
+        explicit Validator(
+            ValidationPassConfiguration const& cfg,
+            bool                               debug,
+            llvm::TargetMachine*               target_machine = nullptr);
+
+        // Default construction not allowed as this leads to invalid configuration of the allocation
+        // managers.
```

Comment:
...but move constructors are OK.

Reply:
Thanks for point this comment out. It is actually wrong and I have updated it: Copy is prohibited by underlying classes whereas move is allowed. Move semantics would not have any effect on the initialization of the module, so it is not a problem from an initialisation point-of-view.